src/components/Guilds/common/SVG/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import styled from 'styled-components';
+
+export type StyledIconProps = {
+  src: React.FC<React.SVGProps<SVGSVGElement>>;
+  margin?: string;
+};
+
+const StyledIconWrapper = styled.div<StyledIconProps>`
+  margin: ${({ margin }) => margin || '0 8px'};
+  & svg {
+    & path {
+      fill: currentColor;
+    }
+  }
+`;
+
+const StyledIcon = React.memo((props: StyledIconProps) => {

```

Comment:
Quick question! Why did you consider memorization here? I don't see any heavy stuff to memorize. Just small jsx.

Reply:
Because it is a generic component it would be designed to be used anywhere an SVG icon is used therefore I would like to avoid re-renders whenever possible.